src/gh/diffCheck/diffCheck/df_joint_detector.py

```diff
@@ -47,42 +47,115 @@ def _assign_ids(self, joint_face_ids):
+
+        # scale the bounding geometry in the longest edge direction by 1.5 from center on both directions
+        rh_Bounding_geometry_center = Bounding_geometry.GetBoundingBox(True).Center
+        edges = Bounding_geometry.Edges
         edge_lengths = [edge.GetLength() for edge in edges]
         longest_edge = edges[edge_lengths.index(max(edge_lengths))]
 
-        rh_OBB_zaxis = rg.Vector3d(longest_edge.PointAt(1) - longest_edge.PointAt(0))
-        rh_OBB_plane = rg.Plane(rh_OBB_center, rh_OBB_zaxis)
-        scale_factor = 0.09
+        rh_Bounding_geometry_zaxis = rg.Vector3d(longest_edge.PointAt(1) - longest_edge.PointAt(0))
+        rh_Bounding_geometry_plane = rg.Plane(rh_Bounding_geometry_center, rh_Bounding_geometry_zaxis)
+        scale_factor = 0.1
         xform = rg.Transform.Scale(
-            rh_OBB_plane,
+            rh_Bounding_geometry_plane,
             1 - scale_factor,
             1 - scale_factor,
             1 + scale_factor
         )
-        rh_OBB.Transform(xform)
+        Bounding_geometry.Transform(xform)
 
         # check if face's centers are inside the OBB
-        faces = {idx: (face, rh_OBB.IsPointInside(rg.AreaMassProperties.Compute(face).Centroid, sc.doc.ModelAbsoluteTolerance, True)) for idx, face in enumerate(self.brep.Faces)}
+        '''
+        the structure of the disctionnary is as follows:
+        {
+            face_id: (face, is_inside)
+            ...
+        }
+            face_id is int
+            face is Rhino.Geometry.BrepFace
+            is_inside is bool
+        '''
+        faces = {}
+        if is_cylinder_beam:
+            for idx, face in enumerate(self.brep.Faces):
+                faces[idx] = (face, face.IsPlanar(1000 * sc.doc.ModelAbsoluteTolerance))
+        else:
+            for idx, face in enumerate(self.brep.Faces):
+                face_centroid = rg.AreaMassProperties.Compute(face).Centroid
+                coord = face.ClosestPoint(face_centroid)
+                projected_centroid = face.PointAt(coord[1], coord[2])
+                faces[idx] = (face, Bounding_geometry.IsPointInside(projected_centroid, sc.doc.ModelAbsoluteTolerance, True))
+
+        # compute the adjacency list of each face
+        adjacency_of_faces = {}
+        '''
+        the structure of the dictionnary is as follows:
+        {
+            face_id: (face, [adj_face_id_1, adj_face_id_2, ...])
+            ...
+        }
+            face_id is int
+            face is Rhino.Geometry.BrepFace
+            adj_face_id_1, adj_face_id_2, ... are int
+        '''
+        for idx, face in faces.items():
+            if not face[1]:
+                continue
+            adjacency_of_faces[idx] = (face[0], [adj_face for adj_face in face[0].AdjacentFaces() if faces[adj_face][1] and faces[adj_face][0].IsPlanar(1000 * sc.doc.ModelAbsoluteTolerance) and adj_face != idx]) # used to be not faces[adj_face][0].IsPlanar(1000 * sc.doc.ModelAbsoluteTolerance)
```

Comment:
I propose two things here:
* first it would be great if you can contain what you added in a function encapsualted inside the DFJointDetector (_my_cylinder_func()) with some docstrings explaining why you need this because it's a bit cumbersome like this.
* second I propose to do these to state clearly that what you added is needed only for cylinder beams right?
```py
joint_face_ids = []
if is_cylinder_beam:
    my_cylinder_treatement_func()  # change name
else:
    faces = {idx: (face, Bounding_geometry.IsPointInside(rg.AreaMassProperties.Compute(face).Centroid, sc.doc.ModelAbsoluteTolerance, True)) for idx, face in enumerate(self.brep.Faces)}
    joint_face_ids = [[key] + [adj_face for adj_face in value[0].AdjacentFaces() if faces[adj_face][1] and adj_face != key] for key, value in faces.items() if value[1]]
```

Reply:
Done: I created two functions, one for each dictionnary: 

```py
faces = _find_joint_faces(self, bounding_geometry)
adjacency_of_faces = _compute_adjacency_of_faces(self, faces)
adjacency_of_faces = diffCheck.df_util.merge_shared_indexes(adjacency_of_faces)
joint_face_ids = [[key] + value[1] for key, value in adjacency_of_faces.items()]
```